### PR TITLE
Fixes #18636 - Remove deprecation warning from --root-password in host create/update

### PR DESCRIPTION
**Pull request by ofedoren**

### CODE REVIEW

**Comment by rahulbajaj0509** [comment]
[test]

File: lib/hammer_cli_foreman/hostgroup.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@ def self.included(base)
       base.option "--parent", "PARENT_NAME",  _("Name of parent hostgroup")
       base.option "--puppet-classes", "PUPPET_CLASS_NAMES", "",
         :format => HammerCLI::Options::Normalizers::List.new
-      base.option "--root-pass", "ROOT_PASSWORD",  _("Root password")
+      base.option "--root-password", "ROOT_PASSWORD",  _("Root password")
```
Comment (mbacovsky) [inline lib/hammer_cli_foreman/hostgroup.rb]: This leads to duplicity in help:
```
   --root-pass ROOT_PASS                               Root password on provisioned hosts
   --root-password ROOT_PASSWORD                       Root password
```
The API defines the option `root-pass` while you are adding here another one.  The bug referenced from this fix talks just about host update/create so I'd suggest not to touch the hostgroup at all. I see the point in having consistency in host and hostgroup params  so I'd also suggest to create new issue for hostgroups to add `root-password` there with deprecation warning on `root-pass` (the same code you've just removed in hosts) and another issue to remove it two releases later.

File: test/unit/hostgroup_test.rb [hunk 2]
```diff
@@ -69,7 +69,7 @@
     context "parameters" do
       it_should_accept "name, parent_id, environment_id, architecture_id, domain_id, puppet_proxy_id, operatingsystem_id and more",
           ["--name=hostgroup", "--parent-id=1", "--environment-id=1", "--architecture-id=1", "--domain-id=1", "--puppet-proxy-id=1",
-            "--operatingsystem-id=1", "--medium-id=1", "--partition-table-id=1", "--subnet-id=1", '--puppet-ca-proxy-id=1', '--puppet-class-ids=1,2', '--root-pass=foreman']
+            "--operatingsystem-id=1", "--medium-id=1", "--partition-table-id=1", "--subnet-id=1", '--puppet-ca-proxy-id=1', '--puppet-class-ids=1,2', '--root-password=foreman']
```
Comment (mbacovsky) [inline test/unit/hostgroup_test.rb]: No changes in hostgroup yet

File: test/unit/hostgroup_test.rb [hunk 3]
```diff
@@ -83,7 +83,7 @@
     context "parameters" do
       it_should_accept "name, parent_id, environment_id, architecture_id, domain_id, puppet_proxy_id, operatingsystem_id and more",
           ["--id=1 --name=hostgroup2 --title=default/hostgroup2", "--parent-id=1", "--environment-id=1", "--architecture-id=1", "--domain-id=1", "--puppet-proxy-id=1",
-            "--operatingsystem-id=1", "--medium-id=1", "--partition-table-id=1", "--subnet-id=1", '--puppet-ca-proxy-id=1', '--puppet-class-ids=1,2', '--root-pass=foreman']
+            "--operatingsystem-id=1", "--medium-id=1", "--partition-table-id=1", "--subnet-id=1", '--puppet-ca-proxy-id=1', '--puppet-class-ids=1,2', '--root-password=foreman']
```
Comment (mbacovsky) [inline test/unit/hostgroup_test.rb]: no changes in hostgroup yet

**Comment by ofedoren** [comment]
Updated

**Comment by tstrachota** [comment]
Could you please create and issue for hostgroups as suggested in https://github.com/theforeman/hammer-cli-foreman/pull/336#discussion_r147587766 prior to merging this PR?

**Comment by tstrachota** [comment]
@mbacovsky what's the status here, please?
@ofedoren have you opened the issue about deprecating the options in hostgroup commands?

**Comment by tstrachota** [comment]
[test]

**Comment by ofedoren** [comment]
Updated. I've opened the issues as you suggested, hope I'm not mistaken:

http://projects.theforeman.org/issues/22573
http://projects.theforeman.org/issues/22574

**Comment by tstrachota** [comment]
[test]

**Comment by tstrachota** [comment]
The issues seem correct to me. Thank you!
@mbacovsky any other objections or is it ok to merge?

**Comment by mbacovsky** [comment]
:+1: LGTM now. Thanks @ofedoren!

**Pull request merged**